src/differentials.jl

```diff
@@ -227,3 +245,62 @@ end
 Base.conj(x::Thunk) = @thunk(conj(extern(x)))
 
 Base.show(io::IO, x::Thunk) = println(io, "Thunk($(repr(x.f)))")
+
+"""
+    InplaceableThunk(val::Thunk, add!::Function)
```

Comment:
Sorry for being so pedantic about names, but `InplaceableThunk` just reads a little bit clunky for me :upside_down_face: . What about just `InplaceThunk`?

Reply:
Problem is it isn't always inplace.
Only of you do `accumulate!` is it inplace.

Reply:
Even if it isn't always inplace, I would say `InplaceThunk` still would signify that it can be used inplace, but doesn't necessarily mean that it has to. Is `Thunk!` punning too hard?